src/main/scala/uk/gov/hmrc/play/json/Union.scala

```diff
@@ -19,15 +19,15 @@ package uk.gov.hmrc.play.json
 import play.api.libs.json._
 import scala.reflect.ClassTag
 
-class Union[A](typeField: String, readWith: PartialFunction[String, JsValue => JsResult[A]], writeWith: PartialFunction[A, JsValue]) {
+class Union[A](typeField: String, readWith: PartialFunction[String, JsValue => JsResult[A]], writeWith: PartialFunction[A, JsObject]) {
 
-  def and[B](typeTag: String)(implicit ev: B <:< A, ct: ClassTag[B], f: Format[B]) = {
+  def and[B <: A](typeTag: String)(implicit ct: ClassTag[B], f: OFormat[B]) = {
     val readCase: PartialFunction[String, JsValue => JsResult[A]] = {
       case `typeTag` => jsValue: JsValue => Json.fromJson(jsValue)(f).asInstanceOf[JsResult[A]]
```

Comment:
If we were using Scala 2.13 we could use [substituteCo](https://www.scala-lang.org/api/2.13.6/scala/$less$colon$less.html#substituteCo[F[+_]](ff:F[From]):F[To]) to get rid of this cast though